File: go/binary_search/iterative_binary_search.go
package binarysearch

func IterativeBinarySearch(vec []int64, v int64) int {
	left := 0
	right := len(vec) - 1

	for left <= right {
		mid := (left + right) / 2

		if vec[mid] > v {
			right = mid - 1
			continue
		}

		if vec[mid] < v {
			left = mid + 1
			continue
		}

		if vec[mid] == v {
			return mid
		}
	}

	return -1
}

File: rust/src/binary_search/iterative_binary_search.rs
pub fn iterative_binary_search(vec: Vec<i32>, v: i32) -> i32 {
    let mut left = 0;
    let mut right = vec.len() - 1;

    while left <= right {
        let mid = (left + right) / 2;

        if vec[mid] > v {
            right = mid - 1;
            continue;
        }

        if vec[mid] < v {
            left = mid + 1;
            continue;
        }

        if vec[mid] == v {
            return mid as i32;
        }
    }

    -1
}
